clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -215,14 +215,13 @@ def _is_node_safe_to_kill(self, node_ip: str) -> bool:
         return True
 
     def _get_nodes_by_ip(self) -> Mapping[str, KubernetesNode]:
-        pool_label_selector = self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
-        pool_nodes = self._core_api.list_node().items
-
-        return {
-            get_node_ip(node): node
-            for node in pool_nodes
-            if not self.pool or node.metadata.labels.get(pool_label_selector, None) == self.pool
-        }
+        # TODO(CLUSTERMAN-659): Switch to using just pool_label_key once the new node labels are applied everywhere
+        node_label_selector = self.pool_config.read_string(
+            "node_label_key", default=self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
+        )
```

Comment:
nit, actually this doesn't really make things much different, as right now the logic is already "get all nodes, and then filter the ones with pool_label_key==self.pool", so I'd say there's no real added risk in just using `pool_label_key` directly.

Reply:
ignore me, I should read tickets first.